src/app/file-manager/services/s3-export.service.ts

```diff
@@ -0,0 +1,47 @@
+import { Injectable } from '@angular/core';
+import * as Transferables from '../actions/transferables.actions';
+import { Item } from '@app/file-manager/models/item';
+import { SecurityService } from '@app/file-manager/services/security.service';
+import { ElectronIpcApiService } from '@app/file-manager/services/electron-ipc.api.service';
+import { AppState } from '../dbstate/app-state';
+import { Store } from '@ngrx/store';
+
+@Injectable()
+export class S3ExportService {
+
+  constructor(
+    private store: Store<AppState>,
+    private electronService: ElectronIpcApiService,
+  ) { }
+
+  public startFileExportToS3(item: Item) {
+    const dataTransfer = {
+      'gcsToken': SecurityService.getAccessToken(),
+      'bucketName': localStorage.getItem('S3BucketName'),
+      'preserveStructure': localStorage.getItem('preserveStructure'),
+      'item': item
+    };
+    this.electronService.exportS3(dataTransfer);
+  }
+
+  public testCredentials() {
+    const dataTransfer = {
+      'accessKey': SecurityService.getS3AccessKey(),
+      'secretKey': SecurityService.getS3SecretKey(),
+      'bucketName': localStorage.getItem('S3BucketName'),
+    };
+    this.electronService.setCredentials(dataTransfer);
+  }
+
+  public startUpload(item) {
+    const dataTransfer = {
+      'accessKey': SecurityService.getS3AccessKey(),
```

Comment:
Do we need to send AWS credentials both on `test` and on `start upload` methods? I would think once you're authenticated with AWS you wouldn't need to do it again.